util.c

```diff
@@ -1234,3 +1235,58 @@ int set_cpu_affinity(cpuset_t *set)
 }
 #endif
 #endif /* HAVE_CPUSET_T */
+
+void add_cookie_secret(struct nsd* nsd, uint8_t* secret)
+{
+	/* New cookie secret becomes the staging secret (position 1)
+	 * unless there is no active cookie yet, then it becomes the active
+	 * secret.  If the NSD_COOKIE_HISTORY_SIZE > 2 then all staging cookies
+	 * are moved one position down.
+	 */
+	if(nsd->cookie_count == 0) {
+		memcpy( nsd->cookie_secrets->cookie_secret
+		       , secret, NSD_COOKIE_SECRET_SIZE);
+		nsd->cookie_count = 1;
```

Comment:
```suggestion
		nsd->cookie_count = 1;
		memset(secret, 0, NSD_COOKIE_SECRET_SIZE);
```
The erase also happens later, so here too.